### PR TITLE
Us27-28 admin merchant enable/disable

### PR DESCRIPTION
**Pull request by abwmn**
this PR adds enabled attribute via migration to merchant table and checks for it in admin merchants index, separating enabled/disabled merchants into separate sections, with buttons to switch their status via model methods and controller actions.

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -14,15 +14,22 @@ def edit
   def update
     @merchant = Merchant.find(params[:id])
 
-    if @merchant.update(merchant_params)
+    if params[:switch_enabled]
+      @merchant.switch_enabled
+      redirect_to admin_merchants_path, notice: "#{@merchant.name} is #{@merchant.enabled_text}"
```
Comment (thawley2) [inline app/controllers/admin/merchants_controller.rb:19]: I like the error message directly showing what is missing.

File: app/models/merchant.rb [hunk 2]
```diff
@@ -27,4 +27,24 @@ def disabled_items
...
+
+  def self.enabled
+    where(enabled: :true)
+  end
+
+  def self.disabled
+    where(enabled: :false)
+  end
+
+  def button_text
```
Comment (thawley2) [inline app/models/merchant.rb:43]: Future refactor, making a partial for the button and or form.

File: app/views/admin/merchants/index.html.erb [hunk 3]
```diff
@@ -1,7 +1,19 @@
 <h1>Merchants Index</h1>
 
-<div class="merchant_names">
-  <% @merchants.each do |merchant| %>
-    <%= link_to merchant.name, admin_merchant_path(merchant) %><br>
+<div class="enabled_merchants">
+  <% @merchants.enabled.each do |merchant| %>
+    <div class="merchant<%= merchant.id%>">
+      <%= link_to merchant.name, admin_merchant_path(merchant) %><br>
+      <%= button_to merchant.button_text, admin_merchant_path(merchant, switch_enabled: true), method: :patch %>
```
Comment (thawley2) [inline app/views/admin/merchants/index.html.erb:7]: Potential partial refactor.

File: spec/features/admin/merchants/index_spec.rb [hunk 4]
```diff
@@ -25,6 +22,30 @@
         click_link(merchant1.name)
         expect(current_path).to eq(admin_merchant_path(merchant1))
       end
+      
+      it 'I see sections for enabled/disabled merchants with enable/disable buttons next to their names' do
+        visit admin_merchants_path
+      
+        within ".disabled_merchants .merchant#{merchant1.id}" do
```
Comment (thawley2) [inline spec/features/admin/merchants/index_spec.rb:29]: Great use of within blocks

File: spec/models/merchant_spec.rb [hunk 5]
```diff
@@ -52,5 +52,23 @@
...
+      @merchant.switch_enabled
+      expect(!@merchant.enabled?)
+      @merchant.switch_enabled
+      expect(@merchant.enabled?)
+    end
+  end
+
+  describe 'model methods' do
+    it '.enabled' do
+      expect(Merchant.enabled.none?{|mer| mer.enabled == false})
```
Comment (thawley2) [inline spec/models/merchant_spec.rb:67]: Clever

**Review by thawley2 — APPROVED** [review]
This all looks great.  Left some comments for potential future refactors

**Review by sicknacco — APPROVED** [review]
Looks great!

**Pull request merged**